pkg/scheduler/framework/v1alpha1/interface.go

```diff
@@ -170,6 +170,19 @@ type ScorePlugin interface {
 	Score(pc *PluginContext, p *v1.Pod, nodeName string) (int, *Status)
 }
 
+// NormalizeScorePlugin is an interface that must be implemented by "normalize score"
+// plugins to normalize the node scoring results produced by the same plugin's "Score"
+// method.
+type NormalizeScorePlugin interface {
```

Comment:
Since now they are merged, how about we name this interface ScoreWithNormalizePlugin since it is not just normalize.

Reply:
This is a good idea.
I would also like to discuss another idea: can we just have a single `ScorePlugin` interface which has both `Score` and `NormalizeScore` methods. If `NormalizeScore` is not needed, it can be implemented as a noop. I think this simplifies the documentation as well as the implementation.

Reply:
We can still keep both Score and NormalizeScore interfaces. I think the real problem here is that NormalizeScore is really part of Score plugin. However, in the [plugin config](https://github.com/kubernetes/kubernetes/blob/2cd5fc54a10623c30b257f904244d560f1fad9a0/pkg/scheduler/apis/config/types.go#L165-L169), we have both Score and NormalizeScore PluginSets. 

My suggestion is to have only Score PluginSet in the config. In the framework init phase, we check if the enabled Score plugin also implements NormalizeScore, if so, we add it to framework's [normalizeScorePlugins](https://github.com/kubernetes/kubernetes/blob/26df547c414d87861e197855c65ee6fb7493d93a/pkg/scheduler/framework/v1alpha1/framework.go#L45).

I prefer to keep this PR as is but I will have a followup PR to implement my suggestion if it sounds good to everyone.

Reply:
I think removing the explicit [NormalizeScore list](https://github.com/kubernetes/kubernetes/blob/2cd5fc54a10623c30b257f904244d560f1fad9a0/pkg/scheduler/apis/config/types.go#L169) from the config makes sense if we agree that we don't need to allow disabling NormalizeScore for a plugin that implements Score and NormalizeScore.

Doing it in a separate PR is fine, though I prefer to do it in this one if there is no objection to the assumption stated above, @bsalamat WDYT?

Reply:
@ahg-g +1

Reply:
NormalizeScore list is still in the API but not used in the code. I am going to remove it from the API in a separate PR. Resolving this discussion.